valkey/client.py

```diff
@@ -404,12 +404,27 @@ def transaction(
         Convenience method for executing the callable `func` as a transaction
         while watching all keys specified in `watches`. The 'func' callable
         should expect a single argument which is a Pipeline object.
+
+        :param watch_delay: This keyword-only argument lets you specify a
+            delay time in seconds after a `WatchError` before the transaction
+            is retried. Default is no delay.
+        :param max_tries: This keyword-only argument lets you specify the
+            maximum number to times the transaction should be retried. If the
+            limit is reached, a `ValkeyError` is raised. Default is an
+            **infinite** number of retries!
         """
         shard_hint = kwargs.pop("shard_hint", None)
         value_from_callable = kwargs.pop("value_from_callable", False)
         watch_delay = kwargs.pop("watch_delay", None)
+        max_tries = kwargs.pop("max_tries", None)
         with self.pipeline(True, shard_hint) as pipe:
+            tries = 0
             while True:
+                tries += 1
+                if max_tries and max_tries > 0 and tries > max_tries:
```

Comment:
```suggestion
                tries += 1
                if max_tries and tries > max_tries:
```

0 is falsy and doesn't need separate check

Reply:
This particular change would make it so negative value could be passed to an unexpected behaviour. Maybe something like this?

```python
if max_tries is not None and max_tries > 0 and tries > max_tries
```

I would also expect `max_tries=0` work as "repeat infinitely"

Reply:
what would be the bahaviour if negetive is passed?

Reply:
Ah, my bad, I misread the code. I thought it would go from negative number until it reaches 0. But actually it would loop indefinitely, which would be something I would expect here as well. So I guess it's fine :)

Reply:
I meant applying Amirreza's, yes

Reply:
Since I agree with the proposal to change `if max_tries and max_tries > 0 and tries > max_tries:` to `if max_tries and tries > max_tries:` (causing negative `max_tries` to be an immediate error), and my second commit does that, so I'll mark this as resolved for now.